Breaks a test.

```
        Test Additional - #2849 spreadsheet/poi-fuzz.xls HPSF FAILED

        org.opentest4j.AssertionFailedError: spreadsheet/poi-fuzz.xls - failed for handler HPSFFileHandler:  expected class org.apache.poi.util.RecordFormatException ==> Unexpected exception type thrown, expected: <org.apache.poi.util.RecordFormatException> but was: <java.lang.ArithmeticException>
            at app/org.apache.poi.stress@6.0.0-SNAPSHOT-tests/org.apache.poi.stress.TestAllFiles.verify(TestAllFiles.java:296)
        Caused by: java.lang.ArithmeticException: long overflow
            at org.apache.poi.stress@6.0.0-SNAPSHOT-tests/org.apache.poi.stress.TestAllFiles.lambda$handleAdditional$2(TestAllFiles.java:272)
```

--- a/poi/src/main/java/org/apache/poi/hpsf/Array.java
+++ b/poi/src/main/java/org/apache/poi/hpsf/Array.java
@@ -19,6 +19,7 @@ package org.apache.poi.hpsf;
 import org.apache.poi.util.IOUtils;
 import org.apache.poi.util.Internal;
 import org.apache.poi.util.LittleEndianByteArrayInputStream;
+import org.apache.poi.util.RecordFormatException;
 
 @Internal
 public class Array {
@@ -71,9 +72,14 @@ public class Array {
         long getNumberOfScalarValues() {
             long result = 1;
             for ( ArrayDimension dimension : _dimensions ) {
-                // Math.multiplyExact rejects compounding dimensions that would silently
-                // overflow long math and bypass the > Integer.MAX_VALUE guard in Array.read.
-                result = Math.multiplyExact(result, dimension._size);
+                try {
+                    // Math.multiplyExact rejects compounding dimensions that would silently
+                    // overflow long math and bypass the > Integer.MAX_VALUE guard in Array.read.
+                    result = Math.multiplyExact(result, dimension._size);
+                } catch (ArithmeticException e) {
+                    throw new RecordFormatException(
+                            "Overflow when calculating the number of scalar values", e);
+                }
             }
             return result;
         }

--- a/poi/src/main/java/org/apache/poi/hpsf/UnicodeString.java
+++ b/poi/src/main/java/org/apache/poi/hpsf/UnicodeString.java
@@ -28,6 +28,7 @@ import org.apache.poi.util.Internal;
 import org.apache.poi.util.LittleEndian;
 import org.apache.poi.util.LittleEndianByteArrayInputStream;
 import org.apache.poi.util.LittleEndianConsts;
+import org.apache.poi.util.RecordFormatException;
 import org.apache.poi.util.StringUtil;
 
 @Internal
@@ -36,11 +37,20 @@ public class UnicodeString {
 
     private byte[] _value;
 
+    /**
+     * @param lei a LittleEndianByteArrayInputStream
+     * @throws RecordFormatException if there is a problem with calculated length
+     */
     public void read(LittleEndianByteArrayInputStream lei) {
         final int length = lei.readInt();
         // Math.multiplyExact rejects crafted lengths that would silently wrap signed-int math
         // (e.g. length == 0x40000001 -> length*2 == 0x80000002 wraps to negative).
-        final int unicodeBytes = Math.multiplyExact(length, 2);
+        final int unicodeBytes;
+        try {
+            unicodeBytes = Math.multiplyExact(length, 2);
+        } catch (ArithmeticException e) {
+            throw new RecordFormatException("Invalid unicode length", e);
+        }
         _value = IOUtils.safelyAllocate(unicodeBytes, CodePageString.getMaxRecordLength());
         
         // If Length is zero, this field MUST be zero bytes in length. If Length is

--- a/poi/src/test/java/org/apache/poi/hpsf/TestOverflowHardening.java
+++ b/poi/src/test/java/org/apache/poi/hpsf/TestOverflowHardening.java
@@ -20,6 +20,7 @@ import static org.junit.jupiter.api.Assertions.assertThrows;
 
 import org.apache.poi.util.LittleEndian;
 import org.apache.poi.util.LittleEndianByteArrayInputStream;
+import org.apache.poi.util.RecordFormatException;
 import org.junit.jupiter.api.Test;
 
 /**
@@ -42,7 +43,7 @@ class TestOverflowHardening {
         LittleEndianByteArrayInputStream lei = new LittleEndianByteArrayInputStream(data, 0);
 
         UnicodeString us = new UnicodeString();
-        assertThrows(ArithmeticException.class, () -> us.read(lei));
+        assertThrows(RecordFormatException.class, () -> us.read(lei));
     }
 
     /**
@@ -60,15 +61,19 @@ class TestOverflowHardening {
         // 3 dimensions of size 0x80000000 each -> product overflows 2^63 (long max)
         byte[] data = new byte[4 + 4 + 3 * (4 + 4)];
         int off = 0;
-        LittleEndian.putInt(data, off, Variant.VT_I4);        off += 4;
-        LittleEndian.putInt(data, off, 3);                    off += 4;
+        LittleEndian.putInt(data, off, Variant.VT_I4);
+        off += 4;
+        LittleEndian.putInt(data, off, 3);
+        off += 4;
         for (int i = 0; i < 3; i++) {
-            LittleEndian.putUInt(data, off, 0x80000000L);     off += 4;
-            LittleEndian.putInt(data, off, 0);                off += 4;
+            LittleEndian.putUInt(data, off, 0x80000000L);
+            off += 4;
+            LittleEndian.putInt(data, off, 0);
+            off += 4;
         }
         LittleEndianByteArrayInputStream lei = new LittleEndianByteArrayInputStream(data, 0);
 
         Array a = new Array();
-        assertThrows(ArithmeticException.class, () -> a.read(lei));
+        assertThrows(RecordFormatException.class, () -> a.read(lei));
     }
 }

--- a/test-data/poi-integration-exceptions.csv
+++ b/test-data/poi-integration-exceptions.csv
@@ -100,7 +100,7 @@ spreadsheet/sample.strict.xlsx,extract,OPC,,org.apache.poi.ooxml.POIXMLException
 spreadsheet/57914.xlsx,"handle,extract",XSSF,,org.apache.poi.ooxml.POIXMLException,"Strict OOXML isn't currently supported, please see bug #57699",
 spreadsheet/57914.xlsx,extract,OPC,,org.apache.poi.ooxml.POIXMLException,"Strict OOXML isn't currently supported, please see bug #57699",
 spreadsheet/poi-fuzz.xls,additional,HSSF,,org.apache.poi.util.RecordFormatException,Not enough data (0) to read requested (4) bytes,
-spreadsheet/poi-fuzz.xls,additional,HPSF,,java.lang.ArithmeticException,long overflow,
+spreadsheet/poi-fuzz.xls,additional,HPSF,,org.apache.poi.util.RecordFormatException,"Tried to allocate an array of length 1,468,570,764, but the maximum length for this record type is 100,000",
 spreadsheet/poi-fuzz.xls,handle,HPSF,,org.opentest4j.AssertionFailedError,expected: <true> but was: <false>,
 openxml4j/ContentTypeHasEntities.ooxml,"handle,extract",OPC,,org.apache.poi.openxml4j.exceptions.InvalidFormatException,Can't read content types part !,
 ddf/Container.dat,handle,HMEF,,java.lang.IllegalArgumentException,"TNEF signature not detected in file, expected 574529400 but got -268435441",

--- a/test-data/poi-integration-exceptions.csv
+++ b/test-data/poi-integration-exceptions.csv
@@ -100,7 +100,7 @@ spreadsheet/sample.strict.xlsx,extract,OPC,,org.apache.poi.ooxml.POIXMLException
 spreadsheet/57914.xlsx,"handle,extract",XSSF,,org.apache.poi.ooxml.POIXMLException,"Strict OOXML isn't currently supported, please see bug #57699",
 spreadsheet/57914.xlsx,extract,OPC,,org.apache.poi.ooxml.POIXMLException,"Strict OOXML isn't currently supported, please see bug #57699",
 spreadsheet/poi-fuzz.xls,additional,HSSF,,org.apache.poi.util.RecordFormatException,Not enough data (0) to read requested (4) bytes,
-spreadsheet/poi-fuzz.xls,additional,HPSF,,org.apache.poi.util.RecordFormatException,"Tried to allocate an array of length 1,468,570,764, but the maximum length for this record type is 100,000",
+spreadsheet/poi-fuzz.xls,additional,HPSF,,java.lang.ArithmeticException,long overflow,
 spreadsheet/poi-fuzz.xls,handle,HPSF,,org.opentest4j.AssertionFailedError,expected: <true> but was: <false>,
 openxml4j/ContentTypeHasEntities.ooxml,"handle,extract",OPC,,org.apache.poi.openxml4j.exceptions.InvalidFormatException,Can't read content types part !,
 ddf/Container.dat,handle,HMEF,,java.lang.IllegalArgumentException,"TNEF signature not detected in file, expected 574529400 but got -268435441",

